query/config.go

```diff
@@ -2,14 +2,65 @@ package query
 
 import (
 	gocontext "context"
+	"database/sql"
 	"fmt"
 
 	"github.com/flanksource/duty/context"
+	"github.com/flanksource/duty/models"
+	"github.com/flanksource/duty/types"
 	"github.com/google/uuid"
 	"github.com/jackc/pgx/v5"
 	"github.com/jackc/pgx/v5/pgxpool"
 )
 
+var allowedColumnFieldsInConfigs = []string{"config_class", "external_id"}
+
+func GetConfigsByIDs(ctx context.Context, ids []uuid.UUID) ([]models.ConfigItem, error) {
+	var configs []models.ConfigItem
+	for i := range ids {
+		config, err := ConfigItemFromCache(ctx, ids[i].String())
+		if err != nil {
+			return nil, err
+		}
+
+		configs = append(configs, config)
+	}
+
+	return configs, nil
+}
+
+func FindConfigs(ctx context.Context, config types.ConfigQuery) ([]models.ConfigItem, error) {
+	return FindConfigsByResourceSelector(ctx, config.ToResourceSelector())
+}
+
+func FindConfigIDs(ctx context.Context, config types.ConfigQuery) ([]uuid.UUID, error) {
+	return FindConfigIDsByResourceSelector(ctx, config.ToResourceSelector())
+}
+
+func FindConfigsByResourceSelector(ctx context.Context, resourceSelectors ...types.ResourceSelector) ([]models.ConfigItem, error) {
```

Comment:
The naming is a bit awkward because we already have `FindConfigIDs` that operates on `types.ConfigQuery`